Dockerfile

```diff
@@ -0,0 +1,346 @@
+
+WORKDIR /tmp/nginx-release-${NGINX_SHORT_VERSION}
+RUN auto/configure \
+          --with-compat \
+      --add-dynamic-module=/tmp/nginx-module-vts-$VTS_VERSION
+RUN make modules
+RUN make install
+RUN cp /tmp/ngx_http_opentracing_module.so /usr/local/nginx/modules/
+#END nginx-mod-builder
+
+### ---------------------------ddev-nginx-prod--------------------------------------
+### Build ddev-nginx (for DDEV-Live) by converting to single layer
+FROM scratch as ddev-nginx-prod
+COPY --from=nginx-base / /
+COPY --from=nginx-mod-builder /usr/local/nginx/modules/ngx_http_opentracing_module.so /usr/lib/nginx/modules/ngx_http_opentracing_module.so
+COPY --from=nginx-mod-builder /usr/local/nginx/modules/ngx_http_vhost_traffic_status_module.so /usr/lib/nginx/modules/ngx_http_vhost_traffic_status_module.so
+ADD /ddev-nginx-prod-files /
+#END ddev-nginx-prod
+
+### ---------------------------ddev-webserver-base--------------------------------------
+### Build ddev-php-base from ddev-webserver-base
+### ddev-php-base is the basic of ddev-php-prod (for DDEV-Live)
+### and ddev-webserver-* (For DDEV-Local)
+FROM ddev-php-base as ddev-webserver-base
+ENV PHP_VERSIONS="php5.6 php7.0 php7.1 php7.2 php7.3 php7.4"
+ENV BACKDROP_DRUSH_VERSION=1.3.1
+ENV MKCERT_VERSION=v1.4.1
+
+ENV DEBIAN_FRONTEND noninteractive
+ENV TERM xterm
+ENV MH_SMTP_BIND_ADDR 127.0.0.1:1025
+ENV NGINX_SITE_TEMPLATE /etc/nginx/nginx-site.conf
+ENV APACHE_SITE_TEMPLATE /etc/apache2/apache-site.conf
+ENV WEBSERVER_DOCROOT /var/www/html
+# For backward compatibility only
+ENV NGINX_DOCROOT $WEBSERVER_DOCROOT
+ENV TERMINUS_CACHE_DIR=/mnt/ddev-global-cache/terminus/cache
+
+# Defines vars in colon-separated notation to be subsituted with values for NGINX_SITE_TEMPLATE on start
+# NGINX_DOCROOT is for backward compatibility only, to break less people.
+ENV NGINX_SITE_VARS '$WEBSERVER_DOCROOT,$NGINX_DOCROOT'
+ENV APACHE_SITE_VARS '$WEBSERVER_DOCROOT'
+
+ENV CAROOT /mnt/ddev-global-cache/mkcert
+
+RUN wget -q -O /tmp/nginx_signing.key http://nginx.org/keys/nginx_signing.key && \
+    apt-key add /tmp/nginx_signing.key && \
+    echo "deb http://nginx.org/packages/debian/ $(lsb_release -sc) nginx" > /etc/apt/sources.list.d/nginx.list
+
+RUN apt-get update && apt-get -qq install --no-install-recommends --no-install-suggests -y apache2 libcap2-bin locales-all nginx supervisor
+
+RUN for v in $PHP_VERSIONS; do \
+    apt-get -qq install --no-install-recommends --no-install-suggests -y libapache2-mod-$v || exit $?; \
+done
+
+RUN apt-get -qq autoremove && apt-get -qq clean -y && rm -rf /var/lib/apt/lists/*
+
+# Arbitrary user needs to be able to bind to privileged ports (for nginx and apache2)
+RUN setcap CAP_NET_BIND_SERVICE=+eip /usr/sbin/nginx
+RUN setcap CAP_NET_BIND_SERVICE=+eip /usr/sbin/apache2
+
+ADD ddev-webserver-base-files /
+ADD ddev-webserver-base-scripts /
+# END ddev-webserver-base
+
+### ---------------------------ddev-webserver-prod--------------------------------------
```

Comment:
The ddev-webserver-prod can't actually work yet, as ddev-webserver-dev relies on a start.sh that uses sudo. So anything that uses sudo will have to be replaced with Dockerfile actions, I haven't looked closely, but I think that should work. We'll just set up to run as www-data, since there's a canned user.